### PR TITLE
Change flags decorator constructor

### PR DESCRIPTION
**Pull request by bugagashenkj**

### CODE REVIEW

File: lib/decorators.js [hunk 1]
```diff
@@ -24,8 +24,14 @@ class Flags extends ValuesDecorator {
   }
 }
 
-const createFlags = (...values) => new Flags(values);
-createFlags.of = enumDomain => new Flags(null, enumDomain);
+const createFlags = (...args) => {
+  if (args.length === 1 && typeof args[0] === 'object') {
+    return new Flags({ enumDomain: args[0].enumDomain });
```
Comment (nechaido) [inline lib/decorators.js]: I'd prefer
```suggestion
    return new Flags(args[0]);
```

File: lib/decorators.js [hunk 2]
```diff
@@ -24,8 +24,14 @@ class Flags extends ValuesDecorator {
   }
 }
 
-const createFlags = (...values) => new Flags(values);
-createFlags.of = enumDomain => new Flags(null, enumDomain);
+const createFlags = (...args) => {
```
Comment (nechaido) [inline lib/decorators.js]: Please consider replacing `if` with a ternary operator.

File: lib/decorators.js [hunk 3]
```diff
@@ -14,7 +14,7 @@ class Enum extends ValuesDecorator {}
 const createEnum = (...values) => new Enum(values);
 
 class Flags extends ValuesDecorator {
-  constructor(values, enumDomain) {
+  constructor({ values = [], enumDomain }) {
```
[the remark this answers is not in the document]
Comment (bugagashenkj) [reply lib/decorators.js]: enum is reserved

**Comment by nechaido** [comment]
Please add tests.

File: test/decorators.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  try {
+    schema = await load('test/schemas/decorators', options, config);
+
+    const {
+      FlagsFromValues,
+      FlagsFromEnum,
+      SimpleFlags,
+      FlagsOf,
+    } = schema.schemas[0].definition;
```
Comment (nechaido) [inline test/decorators.js]: I'd prefer to extract them from `schema.domains`.

File: test/decorators.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  default: def,
+  fs: { load },
+} = require('..');
+
+const { options, config } = clone(def);
+
+metatests.test('Decorators / Flags', async test => {
+  let schema;
+
+  try {
```
Comment (nechaido) [inline test/decorators.js]: You can remove `try {} catch () {}` because `metatests` now handles aync functions.

**Pull request merged**